.travis.yml

```diff
@@ -40,14 +43,37 @@ script:
-  # codecov will need a report on to reference in future PRs to the release branch
-  # upload unit test coverage results
+  # these first codecov runs will upload a report associated with the commit set through Travis CI environment variables
+  # use codecov script flags to upload the coverage report for the unit tests
   - bash <(curl -s https://codecov.io/bash) -s target-unit-test-results -F unit_tests
-  # upload e2e test coverage results
-  - if [ "$SHOULD_RUN_E2E" = "true" ]; then bash <(curl -s https://codecov.io/bash) -s target-e2e-test-results -F end_to_end_tests; fi
-# notify slack channel of build status
+  # use codecov script flags to upload the coverage report for the e2e tests
+  - |
+    if [ "$SHOULD_RUN_E2E" = "true" ]; then
+      bash <(curl -s https://codecov.io/bash) -s target-e2e-test-results -F end_to_end_tests;
+    fi
+
+  # run maven-semantic-release to potentially create a new release of datatools-server. The flag --skip-maven-deploy is
+  # used to avoid deploying to maven central. So essentially, this just creates a release with a changelog on github.
+  #
+  # If maven-semantic-release finishes successfully and the current branch is master, upload coverage reports for the
+  # commits that maven-semantic-release generated. Since the above codecov run is associated with the commit that
+  # initiated the Travis build, the report will not be associated with the commits that maven-semantic-release performed
+  # (if it ended up creating a release and the two commits that were a part of that workflow). Therefore, if on master
+  # codecov needs to be ran two more times to create codecov reports for the commits made by maven-semantic-release.
+  # See https://github.com/conveyal/gtfs-lib/issues/193. In order to create reports for both the unit and e2e tsts,
+  # the codecov scripts must be ran twice.
+  #
+  # The git commands get the commit hash of the HEAD commit and the commit just before HEAD.
+  - |
+    semantic-release --prepare @conveyal/maven-semantic-release --publish @semantic-release/github,@conveyal/maven-semantic-release --verify-conditions @semantic-release/github,@conveyal/maven-semantic-release --verify-release @conveyal/maven-semantic-release --use-conveyal-workflow --dev-branch=dev --skip-maven-deploy
+    if [[ "$TRAVIS_BRANCH" = "master" ]]; then
+      bash <(curl -s https://codecov.io/bash) -C "$(git rev-parse HEAD)" -s target-unit-test-results -F unit_tests
+      bash <(curl -s https://codecov.io/bash) -C "$(git rev-parse HEAD^)" -s target-unit-test-results -F unit_tests
+      if [ "$SHOULD_RUN_E2E" = "true" ]; then
+        bash <(curl -s https://codecov.io/bash) -C "$(git rev-parse HEAD)" -s target-e2e-test-results -F end_to_end_tests;
+        bash <(curl -s https://codecov.io/bash) -C "$(git rev-parse HEAD^)" -s target-e2e-test-results -F end_to_end_tests;
+      fi
+    fi
 notifications:
   # notify slack channel of build status
   slack: conveyal:WQxmWiu8PdmujwLw4ziW72Gc
```

Comment:
What happens after an e2e run finishes? Are the results uploaded to Teams? Should we remove this slack notification?

Reply:
This is a standard Travis-CI > Slack notification. It doesn't have additional details about the results of the e2e tests, just the overall build status.